youtube_dl/extractor/brightcove.py

```diff
@@ -508,6 +510,20 @@ def _extract_urls(webpage):
                 'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
                 % (account_id, player_id, embed, video_id))
 
+        # 	<video data-video-id="5320421710001" data-account="245991542" data-player="SJWAiyYWg" data-embed="default" class="video-js" controls itemscope itemtype="http://schema.org/VideoObject">
+        for video in re.findall(r'(?i)(<video[^>]+>)', webpage):
+            attrs = extract_attributes(video)
+                
+            video_id    = attrs.get('data-video-id')
+            account_id  = attrs.get('data-account')
+            player_id   = attrs.get('data-player')
+            embed       = attrs.get('data-embed')
+
+            if video_id and account_id and player_id and embed:
+                entries.append(
+                    'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
+                    % (account_id, player_id, embed, video_id))
+
```

Comment:
Honestly, I'm pretty confused what you're asking for. I agree they ought to be merged, and I'm still not clear why a separate extractor is necessary and why this can't just go in `brightcove.py` and handle `data-brightcove-video-id`.

But here you seem to be suggesting `brightcove.py` should change its detection heuristic to search for the script tag?  That doesn't work for the `bostonglobe.com` case, as far as I can tell.

And as for the 2nd, checking tags, isn't that what the code is doing?
Are you saying this should get folded back into `brightcove.py`'s `_extract_urls()` method, and then `bostonglobe.py` should import that? I don't see how to do that without adding `data-brightcove-video-id` to `brightcove.py`, which is what you want me not to do?

So, I'm sorry, I don't understand what you are asking for here.